chain/client/src/client_actor.rs

```diff
@@ -1465,30 +1465,21 @@ impl ClientActor {
     ///
     /// The selected block will always be the first block on a new epoch:
     /// <https://github.com/nearprotocol/nearcore/issues/2021#issuecomment-583039862>.
-    ///
-    /// To prevent syncing from a fork, we move `state_fetch_horizon` steps backwards and use that epoch.
-    /// Usually `state_fetch_horizon` is much less than the expected number of produced blocks on an epoch,
-    /// so this is only relevant on epoch boundaries.
     fn find_sync_hash(&mut self) -> Result<CryptoHash, near_chain::Error> {
         let header_head = self.client.chain.header_head()?;
-        let mut sync_hash = header_head.prev_block_hash;
-        for _ in 0..self.client.config.state_fetch_horizon {
```

Comment:
The comment at the beginning of the function said it's to handle forks. I can imagine that the head block corresponds to a fork that belongs to a different epoch.
I haven't tested what happens in case of a fork, the worst I can think of is getting stuck for a while until switching to Header Sync and then doing another State Sync.